tests/unit/mainwindow/statusbar/test_zoom.py

```diff
@@ -0,0 +1,50 @@
+"""Test Zoom widget."""
+
+import pytest
+
+from qutebrowser.mainwindow.statusbar.zoom import Zoom
+from typing import Any
+import pytestqt.qtbot
+
+
+@pytest.fixture
+def zoom(qtbot: pytestqt.qtbot.QtBot, config_stub: Any) -> Zoom:
+    """Fixture providing a Zoom widget."""
+    widget = Zoom()
+    qtbot.add_widget(widget)
+    return widget
+
+
+@pytest.mark.parametrize('factor, show, expected', [
+    (1, 'always', '[100%]'),
+    (1.5, 'always', '[150%]'),
+    (2, 'always', '[200%]'),
+    (0.5, 'always', '[50%]'),
+    (0.25, 'always', '[25%]'),
+    (1, 'non-default', ''),
```

Comment:
If you test this case separately after the `.hide()` change, you can reduce the duplication here by doing

```python
@pytest.mark.parametrize("show", ["non-default", "always"])
```

on a second line and all parametrizations here will run twice then.